### PR TITLE
CRPProccessInfo: make CreateInstance return a unique_ptr

### PR DESCRIPTION
**Pull request by lrusak**
I made this change when I was investigating #22185 

since `m_processInfo` is already of type `std::unique_ptr<CRPProcessInfo>` we might as well make `RPProcessInfo::CreateInstance` return the same type.

I also made a change so that `CreateRPProcessControl` uses std::function for definition (more readable IMO).

### CODE REVIEW

File: xbmc/cores/RetroPlayer/process/RPProcessInfo.cpp [hunk 1]
```diff
@@ -64,9 +64,9 @@ CRPProcessInfo::CRPProcessInfo(std::string platformName)
 
 CRPProcessInfo::~CRPProcessInfo() = default;
 
-CRPProcessInfo* CRPProcessInfo::CreateInstance()
+std::unique_ptr<CRPProcessInfo> CRPProcessInfo::CreateInstance()
 {
-  CRPProcessInfo* processInfo = nullptr;
+  std::unique_ptr<CRPProcessInfo> processInfo = nullptr;
```
Comment (garbear) [inline xbmc/cores/RetroPlayer/process/RPProcessInfo.cpp:69]: Drop the ` = nullptr`.

Comment (garbear) [reply xbmc/cores/RetroPlayer/process/RPProcessInfo.cpp:69]: And below change the `processInfo != nullptr` comparison to implicit boolean cast

**Review by garbear — APPROVED** [review]

**Comment by lrusak** [comment]
jenkins build this please

**Comment by lrusak** [comment]
rebased and squashed

**Review by garbear — APPROVED** [review]
Newest state looks good

**Pull request merged**